utilities/transactions/write_unprepared_txn.cc

```diff
@@ -52,10 +53,15 @@ WriteUnpreparedTxn::~WriteUnpreparedTxn() {
     // We should rollback regardless of GetState, but some unit tests that
     // test crash recovery run the destructor assuming that rollback does not
     // happen, so that rollback during recovery can be exercised.
-    if (GetState() == STARTED) {
-      auto s __attribute__((__unused__)) = RollbackInternal();
-      // TODO(lth): Better error handling.
+    if (GetState() == STARTED || GetState() == LOCKS_STOLEN) {
```

Comment:
Why LOCKS_STOLEN is added here?

Reply:
In one of the tests that tests transaction expiration, we were calling `delete` on transactions that have expired. In this case, we do want rollback behaviour, since we're implicitly assuming that expired transactions have rolled back.